crates/polars-core/src/chunked_array/ops/downcast.rs

```diff
@@ -99,7 +99,13 @@ impl<T: PolarsDataType> ChunkedArray<T> {
     #[inline]
     pub(crate) fn index_to_chunked_index(&self, index: usize) -> (usize, usize) {
         if self.chunks.len() == 1 {
-            return (0, index);
+            // SAFETY: chunks.len() == 1 guarantees this is correct.
```

Comment:
I don't understand this? If chunks.len == 1, we must always return 0 as chunk index.

Reply:
@ritchie46 No, if the first chunk has length 20, but we ask for index 30, then that would be index 10 in the second chunk, not an index in the first chunk. By doing this consistently you can remove an extra bounds check in all other places because it means that if `chunk_idx < chunks.len()` then the entire thiing is valid.

Reply:
Right, it is a matter of interpretation where we go oob, in the first or the second chunk.

If you say, it makes assumptions down the line easier, I am good. :+1: